pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -226,3 +204,33 @@ func (r *AtlasDeploymentReconciler) ensureAdvancedOptions(ctx *workflow.Context,
 
 	return nil
 }
+
+func (r *AtlasDeploymentReconciler) getProjectDatabaseUsers(ctx context.Context, projectID string) ([]*akov2.AtlasDatabaseUser, error) {
+	databaseUsers := &akov2.AtlasDatabaseUserList{}
+	err := r.Client.List(ctx, databaseUsers, &client.ListOptions{})
```

Comment:
@helderjs @josvazg question: This will list all users in Kubernetes rather than caching them. What was wrong with the indexers/cache?

Reply:
Without going through all users and filtering them manually, the deployment would not create the secrets for all the users under their project. Basically, independent deployment would update independent users, linked would update only linked. In order to keep everything in sync we have to do that.

Reply:
Sorry, I don't understand. The indexers are being updated every time a DB user resource is observed. What loss of information would we have if using a cache?